src/coredump/coredump.c

```diff
@@ -286,16 +286,21 @@ static int make_filename(const char *context[_CONTEXT_MAX], char **ret) {
         if (!p)
                 return -ENOMEM;
 
+        s = filename_escape(context[CONTEXT_SIGNAL]);
+        if (!s)
+                return -ENOMEM;
```

Comment:
Oooh, it's stored as a number? That's even worse, because signal numbers are not portable between architectures, so if it's copied somewhere else, the number becomes meaningless.